extensions/azure/fcc-node-directory-cosmos/src/test/java/org/eclipse/dataspaceconnector/catalog/node/directory/azure/CosmosFederatedCacheNodeDirectoryTest.java

```diff
@@ -0,0 +1,120 @@
+import com.azure.cosmos.CosmosDatabase;
+import com.azure.cosmos.models.CosmosContainerResponse;
+import com.azure.cosmos.models.CosmosDatabaseResponse;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import net.jodah.failsafe.RetryPolicy;
+import org.eclipse.dataspaceconnector.catalog.node.directory.azure.model.FederatedCacheNodeDocument;
+import org.eclipse.dataspaceconnector.catalog.spi.FederatedCacheNode;
+import org.eclipse.dataspaceconnector.common.annotations.IntegrationTest;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.dataspaceconnector.common.configuration.ConfigurationFunctions.propOrEnv;
+
+@IntegrationTest
+class CosmosFederatedCacheNodeDirectoryTest {
+
+    public static final String REGION = "westeurope";
+    private static final String ACCOUNT_NAME = "cosmos-itest";
+    private static final String DATABASE_NAME = "connector-itest";
+    private static final String CONTAINER_NAME = "CosmosFederatedCatalogNodeStoreTest";
+    private static CosmosContainer container;
+    private static CosmosDatabase database;
+    private final String partitionKey = "testpartition";
+    private CosmosFederatedCacheNodeDirectory store;
+    private TypeManager typeManager;
+
+    @BeforeAll
+    static void prepareCosmosClient() {
+        var key = propOrEnv("COSMOS_KEY", null);
+        if (key != null) {
+            var client = new CosmosClientBuilder()
+                    .key(key)
+                    .preferredRegions(Collections.singletonList(REGION))
+                    .consistencyLevel(ConsistencyLevel.SESSION)
+                    .endpoint("https://" + ACCOUNT_NAME + ".documents.azure.com:443/")
+                    .buildClient();
+
+            CosmosDatabaseResponse response = client.createDatabaseIfNotExists(DATABASE_NAME);
+            database = client.getDatabase(response.getProperties().getId());
+        }
+    }
+
+    @BeforeEach
+    void setUp() {
+        assertThat(database).describedAs("CosmosDB database is null - did something go wrong during initialization?").isNotNull();
+        CosmosContainerResponse containerIfNotExists = database.createContainerIfNotExists(CONTAINER_NAME, "/partitionKey");
+        container = database.getContainer(containerIfNotExists.getProperties().getId());
+        typeManager = new TypeManager();
+        typeManager.registerTypes(FederatedCacheNode.class, FederatedCacheNodeDocument.class);
+        store = new CosmosFederatedCacheNodeDirectory(container, partitionKey, typeManager, new RetryPolicy<>());
+    }
+
+    @Test
+    void create() {
```

Comment:
I might not understand this correctly, but the point of this test is the interaction with the container, since it is an integration test...? The `container` instance  is there to validate the correct functionality of the `store`. otherwise we would use the system-under-test to validate the system-under-test

Reply:
In my opinion, when the SUT is an interface to a persistance system, using SUT to validate itself is actually fine, because it avoid specific logic (e.g. data mapping) to be implemented and mantained in the test. But I am totally fine also with the approach used here.